### PR TITLE
Features/adding peers

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Adding peer list support

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -5,8 +5,14 @@ import {ConfigModule} from './modules/config/config.module';
 import {HealthCheckController} from './controllers/health-check.controller';
 import {BlockScannerModule} from './modules/block-scanner/block-scanner.module';
 import {FriendlyBotModule} from './modules/friendly-bot/friendly-bot.module';
+import {PeerModule} from './modules/peers/peer.module';
+
 @Module({
+<<<<<<< HEAD
```
Comment (evgeny-s) [inline src/app.module.ts:11]: @karthikeyan-cere , fix this please

File: src/modules/peers/peer.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  /**
+   * Fetch the list of peer nodes connected to the network
+   * @param network network string
+   * @returns peer details
+   */
+  public async fetch(network: string): Promise<any> {
+    this.logger.log(`About to fetch node details`);
+    let networkWsUrl;
+    if (network === 'TestNet') {
```
Comment (evgeny-s) [inline src/modules/peers/peer.service.ts:22]: Can we use switch-case here? I think it will be much more clear

Comment (evgeny-s) [reply src/modules/peers/peer.service.ts:22]: Let's create ENUM and put all strings for networks to there.

File: src/modules/peers/peer.service.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    return data;
+  }
+
+  /**
+   * Fetch location of the IP address
+   * @param ip IP address
+   * @returns city and country
+   */
+  public async geoLocation(ip: string): Promise<any> {
+    const response = await (await (Axios.get(`http://ipwhois.app/json/${ip}?objects=country,city`))).data;
```
Comment (evgeny-s) [inline src/modules/peers/peer.service.ts:66]: What are the limitations of this service?

Comment (evgeny-s) [reply src/modules/peers/peer.service.ts:66]: Can we add caching here? At least on memory level.

**Review by evgeny-s — CHANGES_REQUESTED** [review]

File: src/app.module.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/app.module.ts:11]: Fixed.

File: src/modules/peers/peer.service.ts (hunk 2 above)
Comment (karthikeyan-cere) [reply src/modules/peers/peer.service.ts:22]: Done.

File: src/modules/peers/peer.service.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply src/modules/peers/peer.service.ts:66]: We can use 10,000 requests per month for free. I haven't added caching as we will not consume 10k requests in a month.

Comment (evgeny-s) [reply src/modules/peers/peer.service.ts]: ok, I'll add improvement ticket for this.

File: src/modules/peers/peer.service.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        break;
+      case Network.DEV:
+        networkWsUrl = this.configService.get('DEV_WS_URL');
+        break;
+
+      case Network.DEV1:
+        networkWsUrl = this.configService.get('DEV1_WS_URL');
+        break;
+      default:
+        break;
```
Comment (evgeny-s) [inline src/modules/peers/peer.service.ts:35]: Let's throw error here in case we didn't found provided network

**Review by evgeny-s — CHANGES_REQUESTED** [review]

File: src/modules/peers/peer.service.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply src/modules/peers/peer.service.ts:35]: done.

**Review by evgeny-s — APPROVED** [review]

**Pull request merged**